uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/CRM/Tasksassignments/BAO/KeyDates.php

```diff
@@ -20,24 +20,19 @@ public function setUpHeadless() {
   }
   
   public function testDatesForDeletedContactsNotInResult() {
-    $contact = ContactFabricator::fabricate(['birth_date' => '1977-06-27']);
-    ContactFabricator::fabricate(['birth_date' => '1978-06-19']);
-    ContactFabricator::fabricate(['birth_date' => '1982-08-16']);
-
-    $query = '
-      UPDATE civicrm_contact
-      SET is_deleted = 1
-      WHERE id = %1
-    ';
-    $params = [1 => [$contact['id'], 'Int']];
-    CRM_Core_DAO::executeQuery($query, $params);
+    ContactFabricator::fabricate(['birth_date' => '1977-06-27', 'is_deleted' => 1]);
+    $contacts[] = ContactFabricator::fabricate(['birth_date' => '1978-06-19'])['id'];
+    $contacts[] = ContactFabricator::fabricate(['birth_date' => '1982-08-16'])['id'];
 
     $keyDates = civicrm_api3('KeyDates', 'get', [
       'sequential' => 1,
     ]);
 
+    $this->assertEquals(sizeof($contacts), $keyDates['count']);
+
+    // Verify deleted contact is not in result
     foreach ($keyDates['values'] as $currentDate) {
-      $this->assertNotEquals($contact['id'], $currentDate['contact_id']);
+      $this->assertContains($currentDate['contact_id'], $contacts);

```

Comment:
I think it will actually be better to use assertEquals here so as to be more specific.
the contacts array will be something like this `[0 => ['id'=>1, 'name'=>'blabkla'], 1 =>['id' => 2, 'name' =>'blabla'] ]`  and we are validating that it contains the contact_id.  To be safer we can do two assertions
$this->assertEquals($keyDates['values'][0]['contact_id'], $contacts[0]['contact_id'])
$this->assertEquals($keyDates['values'][1]['contact_id'], $contacts[1]['contact_id'])` depending on the order the KeyDates API returns the results.